lib/src/chart/line_chart/line_chart_data.dart

```diff
@@ -846,9 +866,15 @@ class FlDotData with EquatableMixin {
         getStrokeColor,
         getDotColor,
         checkToShowDot,
+        getDotShape,
       ];
 }
 
+enum FlDotDataShape {
```

Comment:
I think enum will restrict us in the future,
because we are trying to make it possible to provide a widget to show,
but with the enum, it's impossible or overkill,
On the other hand, maybe we want to customize the shape, for example, if we decide to have a rounded radius in a square shape enum doesn't let us do that.
My suggestion is to have an abstractClass paradigm like this:

```dart
abstract class FlDotDrawer {
  void draw(Canvas canvas, FlSpot spot, Offset offsetInCanvas);
}

class FlDotCircleDrawer extends FlDotDrawer {

  double size;
  double strokeWidth;
  Color color;

  @override
  void draw(Canvas canvas, FlSpot spot, Offset offsetInCanvas) {
    canvas.drawCircle(offsetInCanvas, size, Paint()..color = color);
  }

}

class FlDotSquareDrawer extends FlDotDrawer {

  Color color;
  double size;
  Radius radius;

  @override
  void draw(Canvas canvas, FlSpot spot, Offset offsetInCanvas) {
    final halfSize = size / 2;
    final left = offsetInCanvas.dx - halfSize;
    final top = offsetInCanvas.dy - halfSize;
    final right = offsetInCanvas.dx + halfSize;
    final bottom = offsetInCanvas.dy + halfSize;
    canvas.drawRRect(RRect.fromLTRBR(left, top, right, bottom, radius), Paint()..color = color);
  }

}
```

then we can have a reference to the `FlDotDrawer` class, and use `FlDotDrawer.draw` to draw it in the line_chart_painter class,
it has lot's of benefits, like:

1. we are not restricted, and we can have any kind of shape that we want, like having cross shape, or anything that we can draw on the canvas.
 
2. each shape properties will be held inside of that class, instead of `FlDotData` (I mean all color, size,... properties should be removed from `FlDotData`, and each concrete class should hold it's own properties) 

3. each paint logic will be handled in the child concrete classes, and our painter class will stay as clear as possible.

Let me know what do you think.

Thanks!

Reply:
I know it is a huge and time-consuming job, but I promise you, it will help people a lot.
and It's a professional way to do that.

It will help us to have any kind of shape that you think.
Thanks in advance!

Reply:
And for sure, it's kind of pseudocode, I mean I'm not sure if is it working or not, I didn't run it.

Reply:
Oh I totally agree. I was actually hoping when I first had the requirement that you were already using a Widget to draw the Dots and it would just be a matter of exposing that widget haha. After I saw the Canvas drawing I went for the enum just because I had an immediate problem to solve for our product, but I totally agree about this. I have some other things to finish first, but I'll try to tackle this asap. Thanks for the pointers and glad to hear you backed the idea

Reply:
Yes, drawing the widget is the next step (It's kinda hard to do that, but we should think about it),
but for now, let's expose the canvas to draw anything we want (or anything end-user wants).
I'm going to bump up the version to 0.10.0,
It will be great if we introduce it in this release,
when do you think you can have it finished?

Reply:
I will make my best to have it done today, if not, tomorrow

Reply:
Thanks!